arch/arc/core/regular_irq.S

```diff
@@ -35,6 +35,168 @@ GTEXT(_rirq_common_interrupt_swap)
  * TODO: Revist this if FIRQ becomes configurable.
```

Comment:
I think this TODO doesn't hold anymore, FIRQ is configurable now. Does it mean the NUM_REGULAR_IRQ_PRIO_LEVELS isn't corrent in that case?

Reply:
I guess this is orthogonal to this particular pull-request and thus should be addressed separately as I'm not touching any topics outside complex return from IRQ.